### ISSUE TITLE
UnknownIndex crash

### ISSUE THREAD

**Issue by santoshakil** [isar/isar#795]
App crashed with this error: **Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 7222 (isarworker), pid 7130 (ation.myapp)**
I tested the app on OnePlus 8. The app works fine on OnePlus 8 but on the Pixel 3a, the app crashes. I tried uninstalling the app and installing it again but the same issue. But, if I clear the app data and open it again then it works fine.

> Build fingerprint: 'google/sargo/sargo:12/SP2A.220505.008/8782922:user/release-keys'
> Revision: 'MP1.0'
> ABI: 'arm64'
> Timestamp: 2022-10-18 11:10:59.803038557+0600
> Process uptime: 0s
> Cmdline: com.example.myapp
> pid: 7130, tid: 7222, name: isarworker  >>> com.example.myapp <<<
> uid: 10349
> signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
> Abort message: 'called `Result::unwrap()` on an `Err` value: UnknownIndex'
>     x0  0000000000000000  x1  0000000000001c36  x2  0000000000000006  x3  000000799807a900
>     x4  0080808080808080  x5  0080808080808080  x6  0080808080808080  x7  8080808080808000
>     x8  00000000000000f0  x9  a175ec62819bae21  x10 0000000000000000  x11 ffffff80fffffbdf
>     x12 0000000000000001  x13 7865646e496e776f  x14 000000799807a808  x15 0000000000000050
>     x16 0000007c3c243050  x17 0000007c3c21fdb0  x18 000000799791a000  x19 0000000000001bda
>     x20 0000000000001c36  x21 00000000ffffffff  x22 00000079dacb8610  x23 0000000000000000
>     x24 0000007938d9f4a0  x25 0000000000000001  x26 0000007938dfa968  x27 00000079cacca338
>     x28 000000000000000b  x29 000000799807a980
>     lr  0000007c3c1d2aa0  sp  000000799807a8e0  pc  0000007c3c1d2acc  pst 0000000000000000
> backtrace:
>       #00 pc 000000000004facc  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: cd7952cb40d1a2deca6420c2da7910be)
>       #01 pc 00000000000b0f20  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #02 pc 00000000000b0e2c  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #03 pc 00000000000d5180  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #04 pc 00000000000d6f24  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #05 pc 00000000000d6adc  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #06 pc 00000000000d6a58  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #07 pc 00000000000d6a24  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #08 pc 0000000000069f68  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #09 pc 000000000006c850  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #10 pc 0000000000094910  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #11 pc 000000000008ded8  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #12 pc 0000000000088a5c  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #13 pc 0000000000062d8c  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so (isar_instance_create+1064)
>       #14 pc 0000000000046080  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #15 pc 00000000000d87dc  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #16 pc 00000000000d7ba8  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #17 pc 00000000000d7778  /data/app/~~ke10v5FfACVt3CjtvL5Hnw==/com.example.myapp-1F_Pv7rngMdDuBFhXwETbw==/lib/arm64/libisar.so
>       #18 pc 00000000000b1810  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+264) (BuildId: cd7952cb40d1a2deca6420c2da7910be)
>       #19 pc 00000000000512f0  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: cd7952cb40d1a2deca6420c2da7910be)
> Lost connection to device.
> Exited

### Details

 - Platform: Google Pixel 3a, Android 12
 - Flutter version: 3.3.4
 - Isar version: 3.0.2

**Comment by openWebX** [comment]
Same Problem here...

```
I/Gralloc4(31060): Adding additional valid usage bits: 0x8202000
F/libc    (31060): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 31175 (isarworker), pid 31060 (loud.vc_flutter)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'Xiaomi/raphael/raphael:12/SQ3A.220705.004/1457:userdebug/release-keys'
Revision: '0'
ABI: 'arm64'
Timestamp: 2022-10-19 14:17:10.616859010+0200
Process uptime: 0s
Cmdline: com.vineyardcloud.vc_flutter
pid: 31060, tid: 31175, name: isarworker  >>> com.vineyardcloud.vc_flutter <<<
uid: 10381
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'called `Result::unwrap()` on an `Err` value: UnknownIndex'
    x0  0000000000000000  x1  00000000000079c7  x2  0000000000000006  x3  00000075994fa900
    x4  0080808080808080  x5  0080808080808080  x6  0080808080808080  x7  8080808080808000
    x8  00000000000000f0  x9  00000078e16630b0  x10 0000000000000000  x11 ffffff80fffffbdf
    x12 0000000000000001  x13 7865646e496e776f  x14 0000000000000006  x15 0000000000000050
    x16 00000078e1701050  x17 00000078e16dddd0  x18 000000758fc84000  x19 0000000000007954
    x20 00000000000079c7  x21 00000000ffffffff  x22 b400007673056b90  x23 0000000000000000
    x24 00000075be15a4a0  x25 0000000000000001  x26 00000075be1b5968  x27 b400007663006648
    x28 000000000000000b  x29 00000075994fa980
    lr  00000078e169092c  sp  00000075994fa8e0  pc  00000078e169095c  pst 0000000000000000
backtrace:
      #00 pc 000000000004f95c  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: 391ee56efb3654310c16bdbb846056eb)
      #01 pc 00000000000b0f20  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #02 pc 00000000000b0e2c  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #03 pc 00000000000d5180  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #04 pc 00000000000d6f24  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #05 pc 00000000000d6adc  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #06 pc 00000000000d6a58  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #07 pc 00000000000d6a24  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #08 pc 0000000000069f68  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #09 pc 000000000006c850  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #10 pc 0000000000094910  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #11 pc 000000000008ded8  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #12 pc 0000000000088a5c  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #13 pc 0000000000062d8c  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk (isar_instance_create+1064)
      #14 pc 0000000000046080  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #15 pc 00000000000d87dc  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #16 pc 00000000000d7ba8  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #17 pc 00000000000d7778  /data/app/~~Hz5_jSOAUGQaqcvDoNefhw==/com.vineyardcloud.vc_flutter-Ug7hL8S2zBYmc_Ovyu3LQQ==/base.apk
      #18 pc 00000000000b1790  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 391ee56efb3654310c16bdbb846056eb)
      #19 pc 00000000000511ac  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 391ee56efb3654310c16bdbb846056eb)
```

**Comment by leisim** [comment]
Does anyone have a reproducible example?

**Comment by AlexisL61** [comment]
Same problem here using an android emulator (API 33)

**Comment by punitganshani** [comment]
Is there any fix for this?

**Comment by punitganshani** [comment]
Never upgraded from 2.5. Directly used 3.x. Tried on a new database too, and same issue persists

**Comment by leisim** [comment]
Found a way to reproduce, working on a fix!

**Comment by punitganshani** [comment]
Fantastic - I suggest you add some telemetry in future to collect diagnostics in runtime, as shipping a reproducible code (from actual production one) is difficult for anyone.

**Comment by leisim** [comment]
That is much easier said than done. Do you have a suggestion how to collect and submit enough diagnostic information to fix unknown issue and not violate users privacy?

**Comment by leisim** [comment]
Just released a new version with the fix

**Comment by santoshakil** [comment]
> Just released a new version with the fix

The problem is fixed after upgrading to isar 3.0.3. Thanks, man.
